services/web/server/src/simcore_service_webserver/garbage_collector_core.py

```diff
@@ -350,17 +354,19 @@ async def _remove_single_orphaned_service(
             # 1. opened project disappeared from redis?
             # 2. something bad happened when closing a project?
 
-            # TODO: logic around save_state is not ideal, but it remains
-            # with the same logic as before until it is properly refactored
+            user_id = int(interactive_service.get("user_id", -1))
```

Comment:
what do you mean? You are a few lines below catching ``UserNotFoundError``!

My point is mainly that if the user doe not exist, there is no need to call ``get_user_role(app, user_id)`` with an invalid user id and catch the ``UserNotFoundError``, right? :-)

Reply:
Then I honestly do not understand the reason behind your suggestion.
Why introduce an extra check, when the the except block will still have to catch it?

At most I can rename `user_id` to `possible_user_id` to hint that it might not exist. 

This `user_id` is also required by `has_permission` which only uses it to format errors, so it's value is of no consequence.

Reply:
@pcrespov drop me a message about what you were trying to say here when you have time. I am really not understanding this suggestion.